### PR TITLE
hinting.ts: Move overlapping hints around

### PR DESCRIPTION
**Pull request by glacambre**
This commit makes Tridactyl move overlapping hints around in order to
try to untangle them. This is pretty much a failure, hints are still
overlapping and when they aren't it's hard to tell what hint belongs to
what element. One possibility is that the magic numbers in this commit
(`repulsionForce`, `attraction` and `velocityLoss`) just need to be
tweaked until a sweet spot is found, but it is more probable that moving
hints around is just a bad idea.

This code might still be useful for an april's fools if we set
`velocityLoss` to 1 though.

The main approach I used was to have hints be attracted to their element
while repelling each other. Is there anything else I should/could have tried?

### CODE REVIEW

**Comment by bovine3dom** [comment]
I don't quite see why this needs to be iterative. You can work out the overlap vector **d** with the numbers you have in the overlapsWith function. Then just move each hint **d/2** in either direction.

Is it because you want to resolve N-way overlaps at once? YouTube does have N-way overlaps on its homepage but your code doesn't seem to resolve them (I'm not seeing any movement in the y-direction, for example).

**Comment by glacambre** [comment]
>Is it because you want to resolve N-way overlaps at once?

Yes and you might also create new overlaps by moving hints around.

**Comment by bovine3dom** [comment]
Marking as blocked because we need to think about this some more. So it's only blocked by us. But most blocked things are.

File: src/content/hinting.ts [hunk 1]
```diff
@@ -270,18 +343,48 @@ class Hint {
...
+
+    private updatePosition() {
+        this.flag.style.cssText = `
+            top: ${this._y}px !important;
+            left: ${this._x}px !important;
+        `
+    }
+
+    public overlapsWith(h: Hint) {
+        if (this.width == 0) this.width = this.flag.getClientRects()[0].width
```
Comment (mozbugbox) [inline src/content/hinting.ts]: I tried a simpler approach by copy & paste some of the code of this pull. And realised here the `h.width/h.height` could be `0` in the `overlapsWith()` call. If that happened, overlap would not be detected.

**Pull request closed without merging**